packages/txservice/src/dispatch.ts

```diff
@@ -570,26 +568,24 @@ export class TransactionDispatch extends ChainRpcProvider {
       if (receipt == null) {
         // Receipt is undefined or null. This should never occur; timeout should occur before this does,
         // as a null receipt indicates 0 confirmations.
-        throw new TransactionServiceFailure("Unable to obtain receipt: ethers responded with null.", {
-          method,
+        throw new TransactionProcessingError(TransactionProcessingError.reasons.NullReceipt, method, {
           chainId: this.chainId,
           receipt,
           transaction: transaction.loggable,
         });
       } else if (receipt.status === 0) {
         // This should never occur. We should always get a TransactionReverted error in this event.
-        throw new TransactionServiceFailure("Transaction was reverted but TransactionReverted error was not thrown.", {
-          method,
+        throw new TransactionProcessingError(TransactionProcessingError.reasons.DidNotThrowRevert, method, {
           chainId: this.chainId,
           receipt,
           transaction: transaction.loggable,
         });
       } else if (receipt.confirmations < 1) {
         // Again, should never occur.
-        throw new TransactionServiceFailure("Receipt did not have any confirmations, should have timed out!", {
-          method,
+        throw new TransactionProcessingError(TransactionProcessingError.reasons.InsufficientConfirmations, method, {
```

Comment:
Not a huge fan of the `TransactionProcessingError` either because there's no distinct type matching the distinct reasons. Willing to let it slide iff it's only used for cases where the transaction mining/confirmation breaks

Reply:
ideally each error that would have a unique code assigned to it, should also have a unique type